prow/io/iterator.go

```diff
@@ -0,0 +1,78 @@
+package io
+
+import (
+	"context"
+	"io"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"gocloud.dev/blob"
+	"google.golang.org/api/iterator"
+)
+
+type ObjectAttributes struct {
+	// Name is the name of the object or directory
+	Name string
+	// ObjName is the last segment of the name in case of an object
+	ObjName string
+	// IsDir is true if the object is a directory
+	IsDir bool
+}
+
+// ObjectIterator iterates through storage objects
+// It returns attr as long as it finds objects. When no
+// objects can be found anymore done is returned as true
+type ObjectIterator interface {
+	Next(ctx context.Context) (done bool, attr ObjectAttributes, err error)
+}
+
+// GCSObjectIterator implements ObjectIterator for GCS
+type GCSObjectIterator struct {
+	Iterator *storage.ObjectIterator
+}
+
+func (g GCSObjectIterator) Next(_ context.Context) (done bool, attr ObjectAttributes, err error) {
```

Comment:
Please remove the named return values, as they are not being used correctly (nor are they necessary here).

* this isn't a `(float, float)` situation like `(lat, lon float)` or `(temp, humidity float)` where the names clarify things
* You aren't even using them, aka
  - inner declaration of `err` on L35
  - never set done
  - do not use attr on L37, L40


This should just be `(bool, ObjectAttributes, error)`

https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

Reply:
Yup you're right. Removed them.

I thought it would be useful, but you're right and I guess the threshold for using named return values should be a lot higher considering how easy it is to introduce bugs when using them.